lib/active_model_serializers/adapter/json_api/deserialization.rb

```diff
@@ -188,7 +188,7 @@ def parse_relationship(assoc_name, assoc_data, options)
             end
 
           polymorphic = (options[:polymorphic] || []).include?(assoc_name.to_sym)
-          hash["#{prefix_key}_type".to_sym] = assoc_data['type'] if polymorphic
+          hash["#{prefix_key}_type".to_sym] = assoc_data['type'].singularize.capitalize if polymorphic
```

Comment:
we probably want to encapsulate this singular.capitalize logic in a transformer cc @remear

Reply:
`assoc_data.fetch('type').underscore.classify`

Using [] is dangerous.